### PR TITLE
Update publish modal schema

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

This PR adds missing parameter in publish modal RJSF schema which tell RJSF to decode input data in URI encoded format and show data in input field in decoded format

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by Yashsharma1911** [comment]
From todays call we had a little confusion if we should accept this change or not, well yes we should accept because, Uzairs proposal was to migrate these schemas to schemas repo which was not related to fixing issue, and my proposal was to fix issue first because we previously used to have these schemas in schemas repo and then migrated to Sistent, now migrating them to again in schemas repo will require more work then which is needed right now

For now (right now) I'm focusing on fixing issue not creating extra work which I'm not sure should be a priority, I understand it is nice to have these schemas in schemas repo however I'm not sure if we should put this on priority right now, if we should do then let me know

**Comment by leecalcote** [comment]
@MUzairS15 @aabidsofi19 thoughts?

File: src/schemas/publishCatalogItem/schema.tsx
Comment (aabidsofi19) [inline src/schemas/publishCatalogItem/schema.tsx:36]: @Yashsharma1911  these are native to rjsf right ? . if not where are we documenting these . we should be using a uiSchema and not putting the presentation specific properties in the data structure schema

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

**Comment by aabidsofi19** [comment]
i think the publish schema should be derived from the schema for input of the publish endpoint .

**Comment by leecalcote** [comment]
A relevant discussion - https://github.com/meshery/meshkit/pull/584#issuecomment-2322800552

File: src/schemas/publishCatalogItem/schema.tsx
Comment (Yashsharma1911) [reply src/schemas/publishCatalogItem/schema.tsx:36]: I agree @aabidsofi19 we can move it to uiSchema, we are tracking them here - https://github.com/layer5io/sistent/blob/master/src/schemas/readme.md#custom-properties

Comment (Yashsharma1911) [reply src/schemas/publishCatalogItem/schema.tsx:36]: Also in RJSF section in Meshmap developer guide

Comment (Yashsharma1911) [reply src/schemas/publishCatalogItem/schema.tsx:36]: Ops sorry, I mean "Kanvas"

**Comment by Yashsharma1911** [comment]
> i think the publish schema should be derived from the schema for input of the publish endpoint . to remove the redundancy of redefining the same properties . for the presentation we should use a separate uiSchema ( that defines ui like the widget to use, grid size etc not the data structure )
> 
> // @leecalcote @Yashsharma1911 @MUzairS15

This makes much more sense, we can update the capability of uiSchema to support this, and one part is to move them in single place, it does make sense for me if these schemas get bundled in schema repo npm package.

**Comment by Yashsharma1911** [comment]
There is issue with prettier, lemme fix it

**Comment by leecalcote** [comment]
@Yashsharma1911 are we reordering fields, too?

**Comment by Yashsharma1911** [comment]
I need approval to merge this PR, if looks good would be nice 

> @Yashsharma1911 are we reordering fields, too?

Ops I missed this, lemme fix it quick

**Review by leecalcote — APPROVED** [review]
LGTM

**Review by vishalvivekm — APPROVED** [review]

**Comment by Yashsharma1911** [comment]
Thankyou for approving and merging my PR @vishalvivekm

**Pull request merged**
